Assets/Scripts/Player/PlayerMain.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections;
+using UnityEngine;
+using Cinemachine;
+
+using Actor;
+
+namespace Player {
+
+    public class PlayerMain : MonoBehaviour, iActor {
+        [SerializeField] CinemachineImpulseSource screenShakeOnDamage;
+        [SerializeField] CinemachineImpulseSource screenShakeOnDeath;
+
+        // props
+        System.Guid guid = new System.Guid(); // this is the unique ID used for comparing enemies, bosses, pickups, destructibles etc.
+
+        // cached
+        Health health;
+
+        void OnEnable() {
+            if (health != null) {
+                health.OnDamageTaken.Subscribe(OnDamageTaken);
+                health.OnDeath.Subscribe(OnDeath);
+            }
+
+            // TODO: REMOVE
+            controller.OnFirePress.Subscribe(OnFirePress);
+            controller.OnMeleePress.Subscribe(OnMeleePress);
```

Comment:
These subscriptions are just to test screen shake.

Pressing fire triggers the onDamageTaken screen shake effect.
Pressing melee triggers the onDeath one.